src/enzyme_ad/jax/Passes/AffineToStableHLORaising.cpp

```diff
@@ -634,6 +672,28 @@ tryRaisingOpToStableHLO(Operation *op, IRMapping &mapping, OpBuilder &builder,
     return success();
   }
 
+  // Inner for loops
+  if (auto forOp = dyn_cast<affine::AffineForOp>(op)) {
+    if (forOp.getNumIterOperands() != 0)
+      return failure();
+
+    auto iv = forOp.getInductionVar();
+    auto range = getIVRange(iv);
+    if (!range.has_value())
+      return failure();
+
+    emitIVToStableHLO(builder, iv, *range, mapping, maps);
+
+    for (auto &innerOp : forOp.getBody()->without_terminator()) {
+      if (tryRaisingOpToStableHLO(&innerOp, mapping, builder, maps).failed())
```

Comment:
I'm not sure this is legal, since this assumes that the affine.for iterations are independent/can be run in parallel?

Shouldn't an affinefor here [unless parallel] be raised to a stablehlo.while loop?

Reply:
I mean just more simply consider the following

pfor arg1
  for arg2
     x[arg1] *= 2

Reply:
we're working on trying to autoparallelize inner affine.for's into affine.parallel's. once this is in we can add support for inner parallel's too, which I think would permit this : https://github.com/EnzymeAD/Enzyme-JAX/pull/429